From: Andy Gospodarek <gospo@redhat.com>
Subject: [RHEL5.1 Patch] ipv6: /proc/net/anycast6 unbalanced inet6_dev refcnt
Date: Wed, 7 Mar 2007 14:15:20 -0500
Bugzilla: 231310
Message-Id: <20070307191519.GE27615@gospo.rdu.redhat.com>
Changelog: [ipv6] /proc/net/anycast6 unbalanced inet6_dev refcnt

commit aa6e4a96e7589948fe770744f7bb4f0f743dddaa
Author: David Stevens <dlstevens@us.ibm.com>

    [IPV6]: /proc/net/anycast6 unbalanced inet6_dev refcnt
    
    Reading /proc/net/anycast6 when there is no anycast address
    on an interface results in an ever-increasing inet6_dev reference
    count, as well as a reference to the netdevice you can't get rid of.
    
    Signed-off-by: David S. Miller <davem@davemloft.net>

---

 anycast.c |    1 +
 1 files changed, 1 insertion(+)

--- linux-2.6.18.x86_64/net/ipv6/anycast.c.orig	2006-09-19 23:42:06.000000000 -0400
+++ linux-2.6.18.x86_64/net/ipv6/anycast.c	2007-03-07 13:59:20.734132000 -0500
@@ -462,6 +462,7 @@ static inline struct ifacaddr6 *ac6_get_
 			break;
 		}
 		read_unlock_bh(&idev->lock);
+		in6_dev_put(idev);
 	}
 	return im;
 }

